docs/content/040-adoption-guides/040-nexus-with-nextjs.md

```diff
@@ -0,0 +1,328 @@
+
+Start by intalling the `graphql-request` library.
+
+```bash
+npm i graphql-request
+```
+
+Create a function called `getServerSideProps` and make a call to our GraphQL API. The returned data can be supplied in the returned object which will make it available as a `prop` in the function that produces the page. Next.js will take care of treating this page as a server-side rendered page.
+
+```ts
+// pages/index.tsx
+
+import { request, gql } from 'graphql-request'
+
+export async function getServerSideProps() {
+  const query = gql`
+    {
+      frameworks {
+        id
+        name
+      }
+    }
+  `
+  const data = await request('http://localhost:3000/api/graphql', query)
+  const { frameworks } = data
+  return {
+    props: {
+      frameworks,
+    },
+  }
+}
+
+export default function Home({ frameworks }) {
+  return (
+    <div>
+      <ul>
+        {frameworks.map((f) => (
+          <li key={f.id}>{f.name}</li>
+        ))}
+      </ul>
+    </div>
+  )
+}
+```
+
+When the page renders, we'll see a list of the frameworks.
+
+![a next.js page rendered with a list of frameworks](/assets/next-nexus-3.png)
+
+**Note:** Using `graphql-request` to make a server-side call for data from our GraphQL API is good for initially fetching the data but it won't do much for us if we want to have client-side GraphQL features such as caching. If we want additional features like those provided by Apollo Client, it won't work to mix the data retrieved in the `getServerSideProps` function with data managed on the client by Apollo.
+
+## Schema and Type Generation
+
+A GraphQL SDL file and Nexus typings can automatically be generated by configuring the `outputs` option in `makeSchema`. In a typical project, we can tell Nexus where to place these files by using `__dirname` to define a path. That might looks something like this:
+
+```ts
+const schema = makeSchema({
+  // ...
+  outputs: {
+    schema: `${__dirname}/generated/schema.graphql`,
+    typegen: `${__dirname}/generated/types.ts`,
+  },
+})
+```
+
+Using `__dirname` in this fashion will not work in a Next.js project. Next.js takes control of `__dirname` and overrides it with special behavior which interferes with our ability to use it as we normally would.
+
+The solution is to use `path.join` and `process.cwd` together to describe where to place the generated files.
+
+In the `schema/index.ts` file, configure `makeSchema` to output the generated SDL and typings using `path.join` and `process.cwd`.
+
+```ts diff
+import { makeSchema } from '@nexus/schema';
+import { Framework, Query } from './Query';
++import path from 'path';
+
+const schema = makeSchema({
+  types: [Framework, Query],
++  outputs: {
++    typegen: path.join(
++      process.cwd(),
++      'generated',
++      'nexus-typegen.ts'
++    ),
++    schema: path.join(
++      process.cwd(),
++      'generated',
++      'schema.graphql'
++    )
+  }
+});
+
+export default schema;
+```
+
```

Comment:
How would the build command work? Before `tsc` runs Nexus needs the typings file to be present. Generally we suggest an npm script like `"build": "ts-node --transpileOnly path/to/schema"`. How should it work in this case? Based on what you've written I think we can do this:

```json
{
  "scripts": {
    "build:nexus-typegen": "ts-node --transpileOnly schema",
    "build": "yarn build:nexus-typegen && next build"
  }
}
```

?